openmdao/solvers/solver.py

```diff
@@ -616,6 +617,19 @@ def _solve(self):
                         stall_count += 1
                         if stall_count >= stall_limit:
                             stalled = True
+                        if stall_count == 3:
+                            self.linesearch.options['print_bound_enforce'] = True
+
+                            msg = ("Your model has stalled three times and may be violating the "
+                                   "bounds. In the future, turn on print_bound_enforce in your "
+                                   "solver options to see. For example:\n"
+                                   "'prob.model.nonlinear_solver.linesearch.options"
+                                   "['print_bound_enforce']=True'. \nThe bound(s) being violated "
+                                   "now are:\n")
+                            simple_warning(msg)
+
+                            self._single_iteration()
+                            self.linesearch.options['print_bound_enforce'] = False
```

Comment:
This line will always turn off `print_bound_enforce` after 3 stalls.  Instead, you should cache the value of that option, and restore it here with the cache.

Reply:
Actually, a better solution is to change the conditional `if stall_count == 3:` into `if stall_count == 3 and self.linesearch.options['print_bound_enforce']:`.  That way you don't do any of this if they have the option turned on, and you don't have to cache it.

Reply:
Fixed